compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2956,7 +2957,14 @@ private void reorderNamedArgs(BLangInvocation iExpr, BInvokableSymbol invokableS
             if (namedArgs.containsKey(param.name.value)) {
                 expr = namedArgs.get(param.name.value);
             } else {
-                expr = getDefaultValueLiteral(param.defaultValue, param.type.tag);
+                if(param.defaultValue == null) {
+                    expr = getNullLiteral();
+                } else {
+                    Object defaultValue = ((DefaultValueLiteral) param.defaultValue).getValue();
```

Comment:
Should we change the type of `BVarSymbol.defaultValue` to `DefaultValueLiteral`?